lib/logstash/inputs/log4j.rb

```diff
@@ -59,7 +57,25 @@ def register
     @logger.info("Log4j input")
   end # def register
 
-  public
+  def run(output_queue)
+    if server?
+      while !stop?
+        Thread.start(@server_socket.accept) do |s|
+          s.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
```

Comment:
This #peer method is also used to add the host to the current event.
https://github.com/logstash-plugins/logstash-input-log4j/pull/16/files#diff-1629122b96171829f0ffbe27b8e5777bR123